src/testshade/rs_simplerend.cpp

```diff
@@ -184,3 +185,124 @@ rs_transform_points(OSL::ShaderGlobals* /*sg*/, OSL::StringParam /*from*/,
 {
     return false;
 }
+
+OSL_RSOP bool
+rs_get_attribute_constant_int(int value, void* result)
+{
+    reinterpret_cast<int*>(result)[0] = value;
+    return true;
+}
+
+OSL_RSOP bool
+rs_get_attribute_constant_string(OSL::StringParam value, void* result)
+{
+    reinterpret_cast<OSL::StringParam*>(result)[0] = value;
+    return true;
+}
+
+OSL_RSOP bool
+rs_get_attribute_constant_float(float value, bool derivatives, void* result)
+{
+    reinterpret_cast<float*>(result)[0] = value;
+    if (derivatives) {
+        reinterpret_cast<float*>(result)[1] = 0.f;
+        reinterpret_cast<float*>(result)[2] = 0.f;
+    }
+    return true;
+}
+
+OSL_RSOP bool
+rs_get_attribute_constant_int2(int value1, int value2, void* result)
+{
+    reinterpret_cast<int*>(result)[0] = value1;
+    reinterpret_cast<int*>(result)[1] = value2;
+    return true;
+}
+
+OSL_RSOP bool
+rs_get_attribute_constant_float2(float value1, float value2, bool derivatives,
+                                 void* result)
+{
+    reinterpret_cast<float*>(result)[0] = value1;
+    reinterpret_cast<float*>(result)[1] = value2;
+    if (derivatives) {
+        reinterpret_cast<float*>(result)[3] = 0.f;
+        reinterpret_cast<float*>(result)[4] = 0.f;
+        reinterpret_cast<float*>(result)[5] = 0.f;
+        reinterpret_cast<float*>(result)[6] = 0.f;
+    }
+    return true;
+}
+
+OSL_RSOP bool
+rs_get_attribute_constant_float4(float value1, float value2, float value3,
+                                 float value4, bool derivatives, void* result)
+{
+    reinterpret_cast<float*>(result)[0] = value1;
+    reinterpret_cast<float*>(result)[1] = value2;
+    reinterpret_cast<float*>(result)[2] = value3;
+    reinterpret_cast<float*>(result)[3] = value4;
+    if (derivatives) {
+        reinterpret_cast<float*>(result)[4]  = 0.f;
+        reinterpret_cast<float*>(result)[5]  = 0.f;
+        reinterpret_cast<float*>(result)[6]  = 0.f;
+        reinterpret_cast<float*>(result)[7]  = 0.f;
+        reinterpret_cast<float*>(result)[8]  = 0.f;
+        reinterpret_cast<float*>(result)[9]  = 0.f;
+        reinterpret_cast<float*>(result)[10] = 0.f;
+        reinterpret_cast<float*>(result)[11] = 0.f;
+    }
+    return true;
+}
+
+OSL_RSOP bool
+rs_get_attribute(void* _sg, const char* _object, const char* _name,
+                 long long _type, bool derivatives, int index, void* result)
+{
+    OSL::ShaderGlobals* sg        = reinterpret_cast<OSL::ShaderGlobals*>(_sg);
+    const OSL::StringParam object = OSL::bitcast<OSL::ustringrep>(_object);
+    const OSL::StringParam name   = OSL::bitcast<OSL::ustringrep>(_name);
+    const OSL::TypeDesc type      = OSL::bitcast<OSL::TypeDesc>(_type);
+
+    const RenderState* rs = reinterpret_cast<RenderState*>(sg->renderstate);
+
```

Comment:
Add comment that we need to keep these here for when we don't know name at at compile time, but if codegen hands us a name known at compile time, we expect dead code elimination to prune this down for each call.